delta/plugins/project-deletion/src/test/scala/ch/epfl/bluebrain/nexus/delta/plugins/projectdeletion/ShouldDeleteProjectSuite.scala

```diff
@@ -0,0 +1,199 @@
+import scala.util.matching.Regex
+
+class ShouldDeleteProjectSuite extends BioSuite {
+
+  test("delete a deprecated project") {
+    assertDeleted(
+      shouldBeDeleted(
+        configWhere(deleteDeprecatedProjects = true),
+        projectWhere(deprecated = true)
+      )
+    )
+  }
+
+  test("not delete a non-deprecated project") {
+    assertNotDeleted(
+      shouldBeDeleted(
+        configWhere(deleteDeprecatedProjects = true),
+        projectWhere(deprecated = false)
+      )
+    )
+  }
+
+  test("not delete a deprecated project if the feature is disabled") {
+    assertNotDeleted(
+      shouldBeDeleted(
+        configWhere(deleteDeprecatedProjects = false),
+        projectWhere(deprecated = true)
+      )
+    )
+  }
+
+  test("delete a project which has been inactive too long") {
+    assertDeleted(
+      shouldBeDeleted(
+        configWhere(idleInterval = 24.hours),
+        projectWhere(updatedAt = TwoDaysAgo, lastEventTime = TwoDaysAgo)
+      )
+    )
+  }
+
+  test("not delete a project which has been updated recently") {
+    assertNotDeleted(
+      shouldBeDeleted(
+        configWhere(idleInterval = 24.hours),
+        projectWhere(updatedAt = ThreeHoursAgo, lastEventTime = TwoDaysAgo)
+      )
+    )
+  }
+
+  test("not delete a project which has recent events") {
+    assertNotDeleted(
+      shouldBeDeleted(
+        configWhere(idleInterval = 24.hours),
+        projectWhere(updatedAt = TwoDaysAgo, lastEventTime = ThreeHoursAgo)
+      )
+    )
+  }
+
+  test("not delete a project if the org/label does not match the inclusion regex") {
+    assertNotDeleted(
+      shouldBeDeleted(
+        configWhere(idleInterval = 24.hours, includedProjects = List("hippocampus.+".r, ".*neuron".r)),
+        projectWhere(updatedAt = ThreeHoursAgo, lastEventTime = ThreeHoursAgo, org = "hippocampus", label = "mouse")
+      )
+    )
+  }
+
+  test("not delete a project if the org/label matches the exclusion regex") {
+    assertNotDeleted(
+      shouldBeDeleted(
+        configWhere(idleInterval = 24.hours, excludedProjects = List("hippocampus.+".r, ".*neuron".r)),
+        projectWhere(updatedAt = ThreeHoursAgo, lastEventTime = ThreeHoursAgo, org = "thalamus", label = "neuron")
+      )
+    )
+  }
+
+  test("not run against already deleted projects") {
+    assertNotDeleted(
+      shouldBeDeleted(
+        configWhere(idleInterval = 24.hours),
+        projectWhere(updatedAt = TwoDaysAgo, markedForDeletion = true)
+      )
+    )
+  }
+}
+
+object ShouldDeleteProjectSuite extends Assertions with BioAssertions {
+  case class ProjectFixture(
+      deprecated: Boolean,
+      updatedAt: Instant,
+      lastEventTime: Instant,
+      org: String,
+      label: String,
+      id: Iri,
+      markedForDeletion: Boolean
+  ) {
+    val resource = {
+      val project = ProjectGen.project(
+        orgLabel = org,
+        label = label,
+        markedForDeletion = markedForDeletion
+      )
+
+      ResourceF[Project](
```

Comment:
I think ProjectGen can generate the `ResourceF`

Reply:
The method which generates the resource works in a strange way, I don't think it's good for this situation

Reply:
To explain the problem with `ProjectGen`:
```
def resourceFor(
    project: Project,
    rev: Int = 1,
    subject: Subject = Anonymous,
    deprecated: Boolean = false,
    markedForDeletion: Boolean = false
): ProjectResource
```
The problem is that `markedForDeletion` is set in Project, so it’s a duplication here and it’s unclear which one would be used. We should change this method to not take that as an argument, but that feels out of scope for this PR